src-tauri/src/backend/notebook.rs

```diff
@@ -6,12 +6,35 @@
 //! [nbformat v4]: https://github.com/jupyter/nbformat/blob/v5.10.4/nbformat/v4/nbformat.v4.schema.json
 
 use std::collections::BTreeMap;
+use std::ops::{Deref, DerefMut};
 
 use serde::{Deserialize, Serialize};
 use serde_json::{Map, Value};
+use ts_rs::TS;
+
+/// A simple newtype around `BTreeMap<String, Value>` so we can give it a TS
+/// type.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, TS)]
+#[ts(type = "Record<string, unknown>")]
+pub struct JsonBTreeMap(pub BTreeMap<String, Value>);
```

Comment:
Can we use the ts-rs `serde-json-impl` serde_json compatibility instead so we don't have to define a newtype?

https://github.com/Aleph-Alpha/ts-rs/blob/0ba56be43af1e438053af2b422a82fcff252f7ef/ts-rs/src/lib.rs#L146